### PR TITLE
Move the package package override checks out of colcon-core

### PR DESCRIPTION
**Pull request by cottsay**
This functionality is being moved to a separate colcon extension, [colcon-override-check](https://github.com/colcon/colcon-override-check).

Reverts part of #449

### CODE REVIEW

**Review by sloretz — APPROVED** [review]
LGTM when the replacement is merged

**Comment by cottsay** [comment]
I'm holding this PR until colcon/colcon-common-extensions#15 is merged, which itself is waiting until I have sufficient privileges to release that package.
Merging this prematurely could mean that either we're without override checks or blocking `colcon-core` releases until `colcon-common-extensions` can be released.

**Pull request merged**
